File: scala-spark/src/main/scala/com/myCompany/ks2/Kata2_1.scala
package com.myCompany.ks2

import com.myCompany.Launch
import org.apache.spark.sql.SparkSession
import java.nio.file.Paths

object Kata2_1 {
  def main(args: Array[String]): Unit = {
    val spark: SparkSession = SparkSession.builder.appName("scala-Basics").master("local[*]")
      .getOrCreate

    // Read File from local filesystem and create an RDD
    val pathToDataFile = Paths.get(Launch.absolut_path_to_directory, "LICENSE").toString
    // Read File from local filesystem and create an RDD
    val data = spark.read.textFile(pathToDataFile).rdd
    // split words against word delimiters
    val elements = data.flatMap(l => l.split("[;\\s]"))
    // map operation to index words
    val paires = elements.map( element => (element, 1))
    //reduce operation to count word
    val counts = paires.reduceByKey((a,b) => a+b)
    // display results
    counts.foreach(println)

    spark.stop()
  }
}

File: java-basics/src/main/java/com/myCompany/ks2/Kata2_1.java
package com.myCompany.ks2;

import com.myCompany.Launch;
import org.apache.spark.api.java.*;
import org.apache.spark.sql.SparkSession;
import scala.Tuple2;

import java.io.IOException;
import java.nio.file.Paths;
import java.util.Arrays;

public class Kata2_1 {
    public static void main(String[] args) throws IOException {
        SparkSession spark = SparkSession.builder().appName("Java-Basics").master("local[*]")
                .getOrCreate();
        // creating java spark context
        JavaSparkContext jsc = new JavaSparkContext(spark.sparkContext());

        // Read File from local filesystem and create an RDD
        String pathToDataFile = Paths.get(Launch.absolut_path_to_directory, "LICENSE").toString();
        JavaRDD<String> data = jsc.textFile(pathToDataFile);
        // split words against word delimiters
        JavaRDD<String> elements = data.flatMap(content -> Arrays.asList(content.split("[;\\s]")).iterator());
        // map operation to index words
        JavaPairRDD<String, Integer> paires = elements.mapToPair(element -> new Tuple2(element, 1));
        //reduce operation to count word
        JavaPairRDD<String, Integer> counts = paires.reduceByKey((a,b) -> a+b);
        // display results
        counts.foreach(l -> System.out.println(l));

        spark.stop();
    }
}
